requirements-dev.txt

```diff
@@ -1,3 +1,4 @@
+ipython==7.16.1
```

Comment:
The Sphinx docs use nbsphinx to build doc versions of the tutorial notebooks. This requirement was added to avoid a long sequence of warning that were produced with the newer version of nbsphinx.